### PR TITLE
feat(skills): add release test scope workflow

### PR DESCRIPTION
**Pull request by vevaidyanath-lang**
## Summary

Adds a reusable release-test-scope skill that derives a cross-functional QA and feature-review artifact from a release branch without treating unrelated main-branch work as release scope. Includes the generated 0.6.0 report as a dated example so developers can review both the workflow and its concrete output.

## Changes

- Add the `release-test-scope` workflow, release-boundary rules, forward-merge verification, and local-only default.
- Add one shared report template for QA, Engineering, PM, TPM, and developer review.
- Add the 0.6.0 output as a non-authoritative example anchored to immutable release and main SHAs.
- Omit approval, decisions-required, and generation-notes sections from generated reports.

## Type of Change

- [ ] Code change (feature, bug fix, or refactor)
- [ ] Code change with documentation updates
- [ ] Documentation only
- [x] Contributor tooling or automation
- [ ] CI, build, or test infrastructure

## Quality Gates

- [ ] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [x] Tests not applicable — justification: Markdown-only skill, template, and example report; the skill validator and repository hooks provide the applicable validation.
- [x] Documentation updated for user-visible behavior
- [ ] Documentation not applicable — justification:

## Verification

- [x] Pull request title follows the repository's Conventional Commit format
- [x] Every commit includes an appropriate `Signed-off-by:` trailer
- [x] `uv run pre-commit run -a` passes, or any blocked checks are identified below
- [x] Targeted tests pass, or tests are marked not applicable above
- [x] No secrets, API keys, or credentials are included

Targeted validation:

- PASS: `.venv/bin/python /Users/vevaidyanath/.codex/skills/.system/skill-creator/scripts/quick_validate.py .agents/skills/release-test-scope`
- PASS: `SKIP=studio-lint-staged flox -q activate -- .venv/bin/pre-commit run -a`
- PASS: `git diff --cached --check`
- PASS: DCO audit for commit `49ff4f333f7108635e4e4f52d3261e81e6ec62b0`.
- BLOCKED (unrelated): the unskipped `studio-lint-staged` hook requires Studio dependencies (`lint-staged` was not installed) in this isolated worktree. No `web/` files are changed.

### CODE REVIEW

File: .agents/skills/release-test-scope/references/0.6.0-example-report.md [hunk 1]
```diff
@@ -0,0 +1,1016 @@
...
+| Release/version snapshot | `origin/release/0.6` (`913cd89c1bce8d494f8b58b25782b09d64fc9e0d`) |
+| Main verification snapshot | `origin/main` (`54c3da490d1976aac7eb8d5cdc468b2a2d51ea62`) |
+| Release/main merge base | `913cd89c1bce8d494f8b58b25782b09d64fc9e0d` |
+| Generated | `2026-09-17` |
+| Review deadline | `2026-09-17` |
+
+## Release collection summary
+
+- Candidate range: `0.5.0..origin/release/0.6` only.
+- 267 release-range commits.
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/references/0.6.0-example-report.md:20]: so LLMs _love_ putting the exact numbers of things, but I find them pretty useless, is there a way we can coax it to not do that?

File: .agents/skills/release-test-scope/references/0.6.0-example-report.md [hunk 2]
```diff
@@ -0,0 +1,1016 @@
...
+- 68 associated PRs touched `docs/**` according to GitHub changed-file metadata.
+- 67 touched substantive documentation outside `docs/about/release-notes/**`; one was release-note-only.
+- 18 consolidated capabilities: 16 `Proposed`, two `Needs clarification`.
+- Forward-merge states: 18 `Exact commit on main`; zero in every other state.
+- Status: `Draft`; this is not an approved or frozen scope.
+- Highest-risk journeys: sandboxed Gym evaluation, agent packaging/deployment/execution, workload credential delegation, remote-only Anonymizer execution, GRPO environment packaging, Studio template provisioning, and Kubernetes/OpenSandbox configuration.
+
+## Release-to-main forward-merge trace
+
+All release-range commits, including each capability's cited release commits, are ancestors of `origin/main` at `54c3da490d1976aac7eb8d5cdc468b2a2d51ea62`.
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/references/0.6.0-example-report.md:31]: do we want this trace in this artifact or separate?

Comment (mckornfield) [reply .agents/skills/release-test-scope/references/0.6.0-example-report.md:31]: this can also probably be generated relatively easily with a script that's checked in, instead of having the LLM do so

File: .agents/skills/release-test-scope/references/0.6.0-example-report.md [hunk 3]
```diff
@@ -0,0 +1,1016 @@
...
+
+Fabric agents can be packaged from source, published to a registry, deployed through subprocess, Docker, or Kubernetes backends, and used through persistent deployment-backed sessions.
+
+**User impact**
+
+Agent builders can move from a source checkout to a registry-published image and stateful chat without maintaining a separate wrapper.
+
+#### Studio validation
+
+Not applicable; the cited page documents CLI and SDK paths.
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/references/0.6.0-example-report.md:179]: this seems to be NA for a lot 🤷 wonder if it shouldn't even be a present section UNLESS there's something to validate

File: .agents/skills/release-test-scope/references/0.6.0-example-report.md [hunk 4]
```diff
@@ -0,0 +1,1016 @@
...
+
+### Agent environments and ETHOS
+
+| Field | Value |
+|---|---|
+| Disposition | `Proposed` |
+| Confidence | `High` |
+| Area | Agents / Optimizer |
+| Surface | CLI / API-SDK |
+| Forward-merge state | `Exact commit on main` |
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/references/0.6.0-example-report.md:275]: If these basically always say "exact commit on main", are they that useful?

File: .agents/skills/release-test-scope/references/ticket-template.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
+<!-- SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved. -->
+<!-- SPDX-License-Identifier: Apache-2.0 -->
+
+# <version> Release Test Scope
+
+> Status: Draft
+>
+| Field | Value |
+|---|---|
+| Previous release | `<previous_ref>` (`<previous_sha>`) |
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/references/ticket-template.md:10]: I went through these in the example a bit, but it might be worth considering which things are really repetitive/never really tell you much and just pruning those

File: .agents/skills/release-test-scope/references/ticket-template.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+| Release/main merge base | `<merge_base_sha>` |
+| Generated | `<timestamp>` |
+| Review deadline | `<timestamp>` |
+| GitHub enrichment | `Complete / Unavailable / Partial` |
+
+## Release collection summary
+
+- Candidate range: `<previous_ref>..<release_ref>`.
+- Release-range commits: `<count>`.
+- Associated PRs: `<count or unavailable>`; commit-only evidence: `<count>`.
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/references/ticket-template.md:22]: I said this above but: how useful are these numbers?

File: .agents/skills/release-test-scope/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+- Do not invent behavior, Studio navigation, CLI commands, expected results, owners, or documentation links.
+- Mark incomplete evidence `Needs clarification` and name the missing evidence.
+- Keep manual additions and exclusions explicit; never silently discard them.
+- Generate locally first. Creating or updating a GitHub issue requires explicit user authorization.
+- Never trigger a release, modify a tag, publish release notes, or alter release branches.
+
+## Required capabilities
+
+- Read access to repository history.
+- Local Git history is sufficient to generate a commit-backed draft.
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/SKILL.md:61]: why is this? if you fetch both branches I think everything is there

File: .agents/skills/release-test-scope/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+If `gh auth status` or API access fails, continue from local Git unless the user explicitly required complete GitHub metadata. Mark GitHub enrichment unavailable, use commit SHAs instead of guessed PR metadata or owners, and identify the affected fields as incomplete. Do not call a locally generated artifact complete when GitHub-only evidence is missing.
+
+## Workflow
+
+### 1. Validate boundaries and topology
+
+Run:
+
+```bash
+git rev-parse --verify "${previous_ref}^{commit}"
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/SKILL.md:74]: some of these things might be good to just run in a script so it's externalized for the agent

File: .agents/skills/release-test-scope/SKILL.md [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+- Release-range commit and PR counts.
+- Docs-touched PR and consolidated feature counts.
+- GitHub enrichment status and release-note coverage counts.
+- Counts by forward-merge state and disposition.
+- Path to the local shared artifact.
+- Missing evidence and review decisions.
+- State that the output is a draft for stakeholder review.
+- GitHub issue URL only when creation or update was authorized.
+
+Do not represent the generated artifact as an approval record; approval and scope finalization happen outside it.
```
Comment (mckornfield) [inline .agents/skills/release-test-scope/SKILL.md:250]: idk what this means as far as approval record. I also don't know how well agents do with negative commands personally

File: .agents/skills/release-test-scope/references/ticket-template.md (hunk 6 above)
Comment (vevaidyanath-lang) [reply .agents/skills/release-test-scope/references/ticket-template.md]: I don't have a strong opinion. ok to remove
